overlord/configstate/configcore/refresh.go

```diff
@@ -27,6 +27,17 @@ import (
 )
 
 func validateRefreshSchedule(tr Conf) error {
+	refreshTimerStr, err := coreCfg(tr, "refresh.timer")
+	if err != nil {
+		return err
+	}
+
+	if refreshTimerStr != "" {
+		if _, err = timeutil.ParseSchedule(refreshTimerStr); err != nil {
+			return err
+		}
+	}
+
 	refreshScheduleStr, err := coreCfg(tr, "refresh.schedule")
```

Comment:
Maybe a small comment here like: `// try legacy refresh.schedule setting if new-style refresh.timer is not set` ?